### DISCUSSION TITLE
Add argument to wait() to tell it wait some time then return

### DISCUSSION THREAD

**Discussion by diverger** [alandefreitas/matplotplusplus#259] [Ideas]
In OpenCV, the function 'imshow()' can accept argument to tell it how long it should show the image. That will give user more ability to control the time to view the image.  With that argument, even the user can make the figure show dynamically just like movies.

**Comment by alandefreitas** [comment]
In that case, you can call `draw()`.

**Reply by diverger** [reply]
I have some questions about 'draw()':

1. The 'draw()' is not a global function, right? The functions I use for now are something play with 'mesh', 'surface', etc, almost all of them are 'standalone' ones. I can't figure out how to get the objects associated with the figure, and call 'draw()'.  If I missed something, please tell me.
2. How long will the 'draw()' function make the figure/plot visible by the user? If it draw the figure then close the window immediately, the user never have chance to view the image.

BTW, thanks for the great work.

**Reply by alandefreitas** [reply]
I'm not sure if there's a free `draw()` function. In any case, you can always get the current figure/axes (`gcf()`,`gca()`) handle and call `draw()` on them. The free functions are just conveniences to call plot member functions for the current plot.

`draw()` will show the image and keep executing the rest of the code, which might close the figure if the program ends. `show()` will block the program until the user closes the window. If you want to `draw()` and wait for a while before closing, then you can just do that directly with `draw()` and `std::this_thread::sleep_for(...)`.

**Reply by diverger** [reply]
Thanks, I tried what you suggested, and it works. But the window flicker badly, any method to make it update only the minimum necessary portion of the figure but not the entire one?

**Reply by alandefreitas** [reply]
There's no solution for that yet. You would just need to call `draw()` less often.
